odl/solvers/mlem/mlem.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Maximum Likelihood Expectation Maximization algorithm."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+__all__ = ('mlem_poisson',)
+
+
+def mlem_poisson(op, x, data, iter=1, noise='poisson', callback=None):
```

Comment:
Why name it `mlem_poisson` and take a parameter `noise='poisson'`, seems like we either:

Name it `mlem` and take a parameter

or

Name it `mlem_poisson` and take no parameter.

I would personally prefer the first option.